pkg/resource/deploy/deployment_executor.go

```diff
@@ -567,6 +568,22 @@ func (ex *deploymentExecutor) refresh(callerCtx context.Context) error {
 			step := NewRefreshStep(ex.deployment, res)
 			steps = append(steps, step)
 			resourceToStep[res] = step
+		} else if ex.deployment.opts.TargetDependents {
```

Comment:
If `targetDependents` is enabled, and we _haven't_ matched the resource, we need to check through all the resource's dependencies in case one of them is targeted. If so, this is a dependent of a target, and it should also be targeted.

Reply:
This needs to be transitive as well. We do that normally by adding things to Targets as we go.

Reply:
Aah, I thought `GetAllDependencies` collected transitive dependencies too - I'll update this + the test